### DISCUSSION TITLE
Is code-server free?

### DISCUSSION THREAD

**Discussion by simonjcarr** [coder/code-server#3102] [Q&A] [answered]
The repository contains an MIT Licence, which I think means I can use the software without charge including reselling my own software that includes code-server, as long as I include the licence. 

When I visit the coder website ( https://coder.com/pricing ), it says that a self-hosted version of code-server is $35 per month per user and there is no mention of a community edition.

Which is correct?

**Comment by nooobcoder** [comment]
@simonjcarr do not compare code-server with coder. Both of them are very different in use case. Coder is an enterprise workspace solution for coders to provide environments to people in an organisation on demand whereas code-server is different. @jsjoeio please have an insight on the licensing part. Being not a part / moderator of this project can't say much about it, but i can explain the difference between the two @simonjcarr

**Comment by bpmct** [comment, marked as answer]
Thanks @nooobcoder for the response! code-server is totally free and under the [MIT License](https://github.com/cdr/code-server/blob/main/LICENSE.txt). Coder is our commercial product for developer teams with additional features (user-management, multiple workspaces, etc) is what is discussed on the pricing page.

We'll be sure to add more details on that pricing page to make sure the difference is clear. Some more details on the comparison can be seen here: https://coder.com/docs/comparison

**Comment by tmikaeld** [comment]
I fell into this trap, please make the difference clear at the top of the repo readme so devs don't start dismissing it due to high cost.

**Reply by lg-nineoneninethreenine** [reply]
Kinda good idea... What do you think @jsjoeio?

**Reply by jsjoeio** [reply]
Interesting! @bpmct what are your thoughts? Should we add something to the `README`?

**Reply by lg-nineoneninethreenine** [reply]
I have a kind of idea. If you notice Adobe Brackets' repo, they made a kind of banner with Markdown to show that Brackets is discontinued. So, we can do the same on the README, above the License header, and maybe add this to the Coder website too. @bpmct what do you think?

**Reply by lg-nineoneninethreenine** [reply]
Kinda like this:
![image](https://user-images.githubusercontent.com/76845820/139378000-cee219ec-c945-4a79-b2be-81ddb586cb37.png)

**Reply by lg-nineoneninethreenine** [reply]
Source code for that:
```markdown
| :warning: On September 1, 2021, Adobe will end support for Brackets. If you would like to continue using, maintaining, and improving Brackets, you may fork the project on [GitHub](https://github.com/adobe/brackets). Through Adobe’s partnership with Microsoft, we encourage users to migrate to [Visual Studio Code](https://aka.ms/brackets-to-vscode), Microsoft’s free code editor built on open source.
| ---
```

**Reply by lg-nineoneninethreenine** [reply]
You can customize that as you want. The best is to remove the Warning emoji, add a redirect to Coder for the difference, and add a small gist of the differences. On Coder.com, add a kind of elaborate(but not too elaborate) difference, and use the Markdown Table syntax, get the Tickmark emoji or something, like how on Zorin.com, they show the features of the Pro version, the Core version and the Lite version, you can also show the features in a simple table.

**Reply by lg-nineoneninethreenine** [reply]
We can do it like this: https://github.com/cdr/code-server/discussions/3102#discussioncomment-1556782

Originally done by @tmikaeld, great job, but I requested a small change that can make users realise that Coder is developed by Coder org for enterprises, whereas code-server, again developed by Coder, is free and is for a single user(or a group if they are collaborating on a secret project.

**Reply by bpmct** [reply]
I'm a huge fan of this idea. Would you be able to submit a pull request for this or would you like help? Want to make sure you get credit for this.

Then we can review the copy in the code review and get feedback from the maintainers :)

**Reply by lg-nineoneninethreenine** [reply]
Not with my laptop yet but will submit the PR by Sunday your time!(Monday in the Eastern Hemisphere)

**Reply by lg-nineoneninethreenine** [reply]
Maybe sooner too, I can use the online GH client.

**Reply by lg-nineoneninethreenine** [reply]
Done, #4419 people! Please mention me if there is any correction, I will correct it as soon as I can.

**Comment by tmikaeld** [comment]
🔔   Code Server is the free and open source single-user variant of [Coder](https://coder.com/), an Enterprise developer workspace platform.

Can be as simple as that.

**Reply by lg-nineoneninethreenine** [reply]
Yeah, that can do inside the banner contents. Thanks a bunch @tmikaeld! Adding this comment link there!

**Reply by lg-nineoneninethreenine** [reply]
But a small change is needed. Here:
> single-user variant of Coder, an Enterprise developer workspace platform.

I recommend changing `an` up there, to `our` since Coder created code-server(the free, libre version) and Coder(Enterprise Developer Workspace for companies)

**Reply by lg-nineoneninethreenine** [reply]
Please do that, looping this comment there.

**Reply by bpmct** [reply]
(also shared above)

I'm a huge fan of this idea. Would you be able to submit a pull request for this or would you like help? Want to make sure you get credit for this.

Then we can review the copy in the code review and get feedback from the maintainers :)

**Reply by tmikaeld** [reply]
Only one problem, adding it at the bottom will be just as bad as not having it at all. 
Developers don't read that far when scanning for what the project is, it should be at the top.

**Reply by lg-nineoneninethreenine** [reply]
@tmikaeld done, visit this: https://github.com/cdr/code-server/pull/4419/commits/97614910071bb42a8b176fef4fbcf62ff7d7a920
